File: src/main/java/kr/devport/api/domain/port/repository/ProjectRepository.java
package kr.devport.api.domain.port.repository;

import kr.devport.api.domain.port.entity.Project;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface ProjectRepository extends JpaRepository<Project, Long>, ProjectRepositoryCustom {
    Optional<Project> findByExternalId(String externalId);

    @Query("""
            select p
            from Project p
            order by p.stars desc, p.fullName asc
            """)
    List<Project> findAllForWikiAdmin();
}

File: port/repository-jpa/src/main/kotlin/kr/devport/api/domain/port/repository/ProjectRepository.kt
package kr.devport.api.domain.port.repository

import kr.devport.api.domain.port.entity.Project
import org.springframework.data.domain.Sort
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.util.Optional

interface ProjectRepository :
    JpaRepository<Project, Long>,
    ProjectRepositoryCustom {
    fun findByExternalId(externalId: String): Optional<Project>

    @Query(
        """
        select p
        from Project p
        order by p.stars desc, p.fullName asc
        """,
    )
    fun findAllForWikiAdmin(): List<Project>

    override fun findAll(sort: Sort): MutableList<Project>
}
